staging/src/k8s.io/apiserver/pkg/admission/attributes.go

```diff
@@ -140,6 +145,73 @@ func (record *attributesRecord) AddAnnotation(key, value string) error {
 	return nil
 }
 
+func (record *attributesRecord) GetReinvocationContext() ReinvocationContext {
+	return record.reinvocationContext
+}
+
+func newReinvocationContext() *reinvocationContext {
+	return &reinvocationContext{reinvokeWebhooksIfNeeded: sets.NewString(), reinvokeWebhooks: sets.NewString()}
+}
+
+type reinvocationContext struct {
+	// isReinvoke is true when admission plugins are being reinvoked
+	isReinvoke bool
+	// lastOutput holds the result of the last webhook admission plugin call
+	lastOutput runtime.Object
+	// reinvokeWebhooksIfNeeded holds the set of webhooks to reinvoke if a later mutation occurs
+	reinvokeWebhooksIfNeeded sets.String
+	// reinvokeWebhooks holds the set of webhooks that should be reinvoked
+	reinvokeWebhooks sets.String
+	// reinvokeInTree indicates in-tree plugins should be reinvoked
+	reinvokeInTree bool
+}
+
+func (rc *reinvocationContext) IsReinvoke() bool {
+	return rc.isReinvoke
+}
+
+func (rc *reinvocationContext) SetIsReinvoke() {
+	rc.isReinvoke = true
+}
+
+func (rc *reinvocationContext) AddReinvokeIfNeeded(webhook string) {
+	rc.reinvokeWebhooksIfNeeded.Insert(webhook)
+}
+
+func (rc *reinvocationContext) ShouldReinvoke() bool {
+	return rc.reinvokeInTree || len(rc.reinvokeWebhooks) > 0
+}
+
+func (rc *reinvocationContext) CheckIfLastOutputChanged(object runtime.Object) {
```

Comment:
nit: document or rename the function to indicate what it does when built-in plugins mutated the object

Reply:
Thanks. Fixed as part of cleanup pass on interface.